src/plotting/animations.jl

```diff
@@ -1,109 +1,98 @@
-    particle_kwargs = NamedTuple(), orbit_kwargs = NamedTuple(), newfig = true,
-    disable_axis = false, deletefigs = true, dpi = 100,
-    figsize = (7.2, 7.2))
-
-    p = copy(par)
-    if newfig == true
-        fig = PyPlot.figure(figsize = figsize)
-        plot(bd; ax = PyPlot.gca())
+# This internal function creates a closure for easy plotting for each particle.
+function setup_animation(p::AbstractParticle, bd:: Billiard, t::AbstractFloat,
+                         ax::PyPlot.PyCall.PyObject, raysplitters = nothing;
+                         dt = 0.01, taillength::Int, pcolor = "k",
+                         tailcolor = "C0")
+
+    # get data to plot
+    x,y,vx,vy = timeseries(p, bd, t, raysplitters, dt = dt)
+
+    # initial plot
+    point = ax[:scatter](x[1], y[1], color = pcolor, s = 30.0, zorder = 2)
+    arrow = ax[:quiver](x[1], y[1], 0.08vx[1], 0.08vy[1], color = pcolor,
+                        angles = "xy", scale = 1, width = 0.005, zorder = 2)
+
+    tail, = ax[:plot](x[1:2], y[1:2], c = tailcolor, zorder = 1)
+
+    # frame counter
+    count = 2
+    
+    function plot_frame()
+        # replot point
+        point[:remove]()
+        point = ax[:scatter](x[count], y[count], color = pcolor, s = 30.0,
+                             zorder = 2)
+
+        # replot arrow
+        arrow[:remove]()
+        arrow = ax[:quiver](x[count], y[count], 0.08vx[count], 0.08vy[count],
+                            color = pcolor, angles = "xy", scale = 1,
+                            width = 0.005, zorder = 2)
+
+        # set tail data
+        @views tail[:set_xdata](x[clamp(count-taillength, 1, count):count])
+        @views tail[:set_ydata](y[clamp(count-taillength, 1, count):count])
+
+        # increment frame counter
+        count += 1
     end
 
-    disable_axis && PyPlot.axis("off")
+    function skip_frame()
+        count += 1
+    end
+    
+    return plot_frame, skip_frame, length(x)
+    
+end
 
-    sleeptime = 1/framerate
-    i=0
-    xdata = Vector{Float64}[]
-    ydata = Vector{Float64}[]
 
-    line, point, quiv = nothing, nothing, nothing
+"""
+    animate_evolution(ps, bd, colnumber [, raysplitters]; kwargs...)
+Animate the evolution of a vector of particles `ps` in billiard `bd` for a 
+total of `t` times. Optionally enable ray-splitting.
 
-    while i < colnumber
+### Keyword Arguments
+  * `dt = 0.01` : Time resolution used for production of time series(see 
+    [`timeseries`](@ref). It is not recommended to increase this value to 
+    preserve the accuracy of the plots.
+  * `frameskip = 5` : The amount of `dt`-steps in between succesive frames
+  * `tailtime = 1.0` : The length of the "tail" trailing the particle in time 
+    units
+"""
+function animate_evolution(ps::AbstractVector{<:AbstractParticle{T}}, bd::Billiard,
+                           t, raysplitters = nothing; dt  = 0.01,
+                           frameskip = 5, tailtime = 1.0, ) where {T}
```

Comment:
I think it is good to keep the old behavior of `animate` with respect to the figure: if it wasn't given a specific `axis`, it was creating a new figure, getting current axis and then also calling `plot(bd)` on it.
```suggestion
                           frameskip = 5, tailtime = 1.0, 
                           ax = (figure(); plot(bd); gca())) where {T}
```